walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,234 @@
+          overlappedSlots.add(overlappedSlot);
+          return overlappedSlots;
+      }
+      
+      boolean inclusive = false;
+      // There are at least two things in timeSlots
+      for (int i = 1; i < timeSlots.size(); i++){
+
+          if (timeSlots.get(i).overlaps(overlappedSlot)){
+              if (timeSlots.get(i).end() == TimeRange.END_OF_DAY) {inclusive = true;}
+              overlappedSlot = overlappedSlot.mergeOverlapped(timeSlots.get(i), inclusive);
+              if (i == timeSlots.size() - 1) {// If this is the last time slot
+                    overlappedSlots.add(overlappedSlot);
+              }
+          } else {
+              overlappedSlots.add(overlappedSlot);
+              overlappedSlot = timeSlots.get(i);
+              if (i == timeSlots.size() - 1) {
+                  overlappedSlots.add(overlappedSlot);
+              }
+          }
+      }
+
+      return overlappedSlots;
+  }
+
+  /**
+  * @param timeSlots an arrayList of TimeRanges that are sorted whose elements do not overlap (has gone through getOverlappedSlots)
+  * @return the other slots in a day
+  */
+  private ArrayList<TimeRange> getInverseSlots(ArrayList<TimeRange> timeSlots){
+       
+       ArrayList<TimeRange> otherSlots = new ArrayList<TimeRange>();
+       if (timeSlots.isEmpty()) {return otherSlots;}
+
+       // Assume that timeSlots only get the most concise slots, and the last slot also ends the latest
+    
+       if (timeSlots.get(0).start() > TimeRange.START_OF_DAY) {
+           otherSlots.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, timeSlots.get(0).start(), false));
+       }
+
+       // All elements in timeSlots do not overlap
+       for (int i = 0; i < timeSlots.size() - 1; i++) {
+           TimeRange otherSlot = TimeRange.fromStartEnd(timeSlots.get(i).end(), timeSlots.get(i+1).start(), false);
+           otherSlots.add(otherSlot);
+       }
+    
+       // The order is reserved
+       if (timeSlots.get(timeSlots.size()-1).end() < TimeRange.END_OF_DAY) {
+           otherSlots.add(TimeRange.fromStartEnd(timeSlots.get(timeSlots.size()-1).end(), TimeRange.END_OF_DAY, true));
+       }
+
+       return otherSlots;
+  }
+  
+  
+  /**
+  * @return a list of intersecting time ranges with primary slots as the starting point. The returned time ranges must satisfy the duration.
+  */
+  private ArrayList<TimeRange> getIntersection(ArrayList<TimeRange> primarySlots, ArrayList<TimeRange> optionalSlots, int duration) {
+      ArrayList<TimeRange> intersectionSlots = new ArrayList<TimeRange>();
+
+      for (int i = 0; i < primarySlots.size(); i++){
+        for (int j = 0; j < optionalSlots.size(); j++) {
+            TimeRange primary = primarySlots.get(i);
+            TimeRange optional = optionalSlots.get(j);
+
+            // If there is no overlap, then doesn't care about optional available slots
+            if (primary.overlaps(optional)) {
+                if (optional.contains(primary)) {
```

Comment:
You have multiple contains() overload functions, it would be great to have more accurate and distinguishing names for them, for example, you can use coversTimeRange(), coversTimePoint() etc